README.md

```diff
@@ -1,38 +1,70 @@
+Using the remote API, you can 
+- Retrieve location information (reverse Geocoding)
+- Retrieve routing information for your trip.
+  
+Then you can store all this information in your local database. 
+
+> **[Neshan API](https://platform.neshan.org/)** is one of the APIs we use to retrieve information. Check out their documentation as well.
 
-as you can see, we have two sources for our data in the application. remote API and Database. with remote API, we can request location information(reverse Geocoding) and also routing information for our trip. and if we want to save our trip, the database does that for us.
+### Repositories
 
-> also, we use **[Neshan API](https://platform.neshan.org/)** to receive these informations. you can check the documentation as well.
+We created three repositories to help us manage **ViewModel** requests. This section is critical because it creates and controls all processes. We receive data and transform it into a standard model that ViewModels can use and parse for **Views**.
 
-after that, we have three repositories that help us to manage these requests from **ViewModels**. this section is important because all of the processes make and control here. so, we receive data and convert that to a model that **ViewModels** can use and parse it for **Views**. in the picture below, you can see the difference between the two of them:
+The following image, shows the difference between the two of them:
 
 ![architecture models](https://user-images.githubusercontent.com/73066290/172200304-0e7baeb1-7ae0-462c-97a8-cea06299aee0.png)
 
-> the left model represents the data that we receive from the server. and the right picture represents the data that we need for our view.
+> The code on the left represents the data structure we receive from the server, and the code on the right indicates the format we need for our view.
 
-in ViewModels, we have a different approach for giving access to the Views. every ViewModel can have two outputs and Views can use those for managing their views and actions. at the first, we have a **Stateflow** that represents the state for the view. this flow contains all of the information that the view needs to show(like MVI approach).
 
-and the second output that works with **Sharedflow** sends some actions that can have a message or not. Views parse these actions and then, do an action in view like showing an error and so on... for example:
+In ViewModels, we have two different approaches to accessing the views. Each ViewModel has two outputs that can be used by Views to manage their views and actions:
+- The first one is a **Stateflow** which represents the state of the view. This flow contains all the information that the view needs to display (MVI approach)
+- The second one works with a **Sharedflow** and sends some actions that can either contain a message or be empty. View, then parse these actions and perform several actions, like displaying an error, for example. 
```

Comment:
I think we use "action" in this sentence a lot. maybe it's good to write it like this:

`...View, then parse these actions and perform several works...`